src/PaperG/FirehoundBlob/ScenarioValidators/UnmanagedFacebookValidator.php

```diff
@@ -0,0 +1,240 @@
+        $validationResult = $validateTargeting[self::VALIDATION_RESULT];
+        $validationMessage = $validateTargeting[self::VALIDATION_MESSAGE];
+
+        $startDate = $facebookBlob->getStartDate();
+        if (!empty($startDate) && !is_numeric($startDate)) {
+            $validationMessage .= "Start date is non-numerical. ";
+            $validationResult = false;
+        }
+
+        $endDate = $facebookBlob->getEndDate();
+        if (!empty($endDate) && !is_numeric($endDate)) {
+            $validationMessage .= "End date is non-numerical. ";
+            $validationResult = false;
+        }
+
+        $budget = $facebookBlob->getBudget();
+        if (!empty($budget) && !$this->validateBudget($budget)) {
+            $validationMessage .= "Budget is invalid. ";
+            $validationResult = false;
+        }
+
+        return [self::VALIDATION_RESULT => $validationResult, self::VALIDATION_MESSAGE => $validationMessage];
+    }
+
+    private function validateTargeting(UnmanagedFacebookBlob $facebookBlob)
+    {
+        $validationResult = true;
+        $validationMessage = "";
+
+        $geographicTargeting = $facebookBlob->getGeographicTargeting();
+        if (!empty($geographicTargeting) && !$geographicTargeting->isValid()) {
+            $validationMessage .= "Geographic targeting is invalid. ";
+            $validationResult = false;
+        }
+
+        $demographicTargeting = $facebookBlob->getDemographicTargeting();
+        if (!empty($demographicTargeting) && !$demographicTargeting->isValid()) {
+            $validationMessage .= "Demographic targeting is invalid. ";
+            $validationResult = false;
+        }
+
+        $audienceTargeting = $facebookBlob->getAudienceTargeting();
+        if (!empty($audienceTargeting) && !$audienceTargeting->isValid()) {
+            $validationMessage .= "Audience targeting is invalid. ";
+            $validationResult = false;
+        }
+
+        return [self::VALIDATION_RESULT => $validationResult, self::VALIDATION_MESSAGE => $validationMessage];
+    }
+
+    private function validateStatus($status) {
+        $validStatuses = [Context::STATUS_ACTIVE, Context::STATUS_INACTIVE];
+        return is_string($status) && in_array(strtolower($status), $validStatuses);
+    }
+
+    /**
+     * @param $facebookAdSets FacebookAdSet []
+     *
+     * @return bool
+     */
+    private function validateAdSets($facebookAdSets)
+    {
+        if (!empty($facebookAdSets)) {
+            foreach ($facebookAdSets as $adSet) {
+                /** @var $adSet FacebookAdSet */
+                if (!$adSet->validate()) {
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * @param $budget
+     *
+     * @return bool
+     */
+    private function validateBudget(Budget $budget)
+    {
+        if ($budget->getBudgetType() != Budget::BUDGET_TYPE_DAILY ||
```

Comment:
Not necessarily. A budget can be valid (with impression type and valid amount) but it's not a valid budget for unmanaged facebook scenario.